src/biogeophys/SurfaceWaterMod.F90

```diff
@@ -472,15 +477,24 @@ subroutine QflxH2osfcSurf(bounds, num_hydrologyc, filter_hydrologyc, &
        c = filter_hydrologyc(fc)
 
        if (h2osfcflag==1) then
-          if (frac_h2osfc_nosnow(c) <= params_inst%pc) then
+		  ! if surface_water_ponding_column(c) > 0, means that the surface water over this column will be kept
+		  if (irrigation_inst%surface_water_ponding_column(c) > 0) then 
+		     frac_infclust=0.0_r8
```

Comment:
- [ ] `frac_infclust` does not seem to be used if surface water ponding is on, suggesting that this change can be reverted.